### PR TITLE
Add more methods to scala-native zio-streams

### PR DESCRIPTION
**Pull request by pablf**
Added some methods of `zio-streams` from `jvm` to `native` version and its corresponding tests. Part of `ZPipeline` and `zio.streams.compression` can't be tested in scala-native because the tests use some non-implemented java classes. As the classes that they depend on are implemented in scala-native, I would say that it's better to keep them in `shared` and then possible  ZIO users test that they behave correctly in their projects, given the experimental character of some parts of scala-native.

### CODE REVIEW

File: streams-tests/jvm-native/src/test/scala/zio/stream/ZStreamPlatformSpecific2Spec.scala [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          test("JavaStream") {
+            trait A
+            trait JavaStreamLike[A] extends java.util.stream.Stream[A]
+            lazy val javaStream: JavaStreamLike[A]        = ???
+            lazy val actual                               = ZStream.from(javaStream)
+            lazy val expected: ZStream[Any, Throwable, A] = actual
+            lazy val _                                    = expected
+            assertCompletes
+          },
+          // test("JavaStreamScoped") {
```
Comment (DamianReeves) [inline streams-tests/jvm-native/src/test/scala/zio/stream/ZStreamPlatformSpecific2Spec.scala:50]: Should this remain?

Comment (DamianReeves) [reply streams-tests/jvm-native/src/test/scala/zio/stream/ZStreamPlatformSpecific2Spec.scala:50]: Oh I see this existed in the code you migrated this from.

**Review by DamianReeves — COMMENTED** [review]
Oh this is awesome. Thanks for doing this.

**Comment by DamianReeves** [comment]
This ends up taking care of #8298 once its merged.

**Comment by DamianReeves** [comment]
Also, though not strictly streams related. Seems like all you ScalaNative work would be a nice basis to enable this one to #8296

File: streams-tests/jvm-native/src/test/scala/zio/stream/ZStreamPlatformSpecific2Spec.scala (hunk 1 above)
Comment (pablf) [reply streams-tests/jvm-native/src/test/scala/zio/stream/ZStreamPlatformSpecific2Spec.scala:50]: Probably not, but I was unsure of its original purpose...

**Comment by pablf** [comment]
Great! Then I'll have a look at it when I have some time

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**